### PR TITLE
EPMLSTRTAI-374: modified removeCases method

### PR DESCRIPTION
**Pull request by Walrus94**
modified removeCases method to require only list of suit ID and case IDs and written tests for this method

### CODE REVIEW

File: src/main/java/com/epam/test_generator/controllers/CaseController.java [hunk 1]
```diff
@@ -114,16 +123,26 @@
...
-                                            @RequestBody @Valid SuitDTO suitDTO) {
+                                            @RequestBody Long[] removeCaseIds) {
+        SuitDTO suitDTO = suitService.getSuit(suitId);
+        if (!suitDTO.getCases().stream().map(CaseDTO::getId).collect(Collectors.toList())
+            .containsAll(
+                Arrays.asList(removeCaseIds))) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
         List<Long> caseIds = suitDTO.getCases().stream().map(CaseDTO::getId)
+            .filter(Arrays.asList(removeCaseIds)::contains)
```
Comment (hdghg) [inline src/main/java/com/epam/test_generator/controllers/CaseController.java]: This check is redundant

Consider following code fragment:

        SuitDTO suitDTO = suitService.getSuit(suitId);
        List<Long> existentSuitCaseIds = suitDTO.getCases().stream()
            .map(CaseDTO::getId)
            .collect(Collectors.toList());
        List<Long> idsToRemove = Arrays.asList(removeCaseIds);

        if (!existentSuitCaseIds.containsAll(idsToRemove)) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
        }
        caseService.removeCases(suitId, idsToRemove);

        return new ResponseEntity<>(HttpStatus.OK);

Not only it avoids unnecessary stream iterations and ArrayList creations, but also makes **if** block self-explanatory.

File: src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java [hunk 2]
```diff
@@ -431,9 +468,63 @@ public void testRemoveCase_return500whenRuntimeException() throws Exception {
...
+            .removeCases(anyLong(), anyListOf(Long.class));
+
+        mockMvc.perform(delete("/suits/" + SIMPLE_SUIT_ID + "/cases")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(mapper.writeValueAsString(CASE_IDS)))
+            .andDo(print())
+            .andExpect(status().isInternalServerError());
+
+        verify(casesService).removeCases(eq(SIMPLE_SUIT_ID), eq(Arrays.asList(CASE_IDS)));
+    }
```
Comment (hdghg) [inline src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java]: Add success and failure cases for **containsAll** check.

File: src/main/java/com/epam/test_generator/controllers/CaseController.java (hunk 1 above)
Comment (Walrus94) [reply src/main/java/com/epam/test_generator/controllers/CaseController.java]: Changed method according to this comment

File: src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java (hunk 2 above)
Comment (Walrus94) [reply src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java]: Added tests that drops on containsAll check, and with duplicate ids

**Pull request merged**